libzfs_core/_libzfs_core.py

```diff
@@ -0,0 +1,659 @@
+                                 filesystem has been modified since the last snapshot
+                                 and ``force`` is `False`.
+    :raises DestinationModified: if a full stream is received and the destination
+                                 filesystem already exists and it does not have any
+                                 snapshots, and ``force`` is `False`.
+    :raises DatasetNotFound: if the destination filesystem and its parent do not exist.
+    :raises DatasetNotFound: if the ``origin`` is not `None` and does not exists.
+    :raises DatasetBusy: if ``force`` is `True` but the destination filesystem could not
+                         be rolled back to a matching snapshot because a newer snapshot
+                         is held and could not be destroyed.
+    :raises DatasetBusy: if another receive operation is being performed on the
+                         destination filesystem and this operation has lost the race.
+    :raises BadStream: if the stream is corrupt or it is not recognized or it is
+                       a compound stream or it is a clone stream, but ``origin``
+                       is `None`.
+    :raises BadStream: if a clone stream is received and the destination filesystem
+                       already exists.
+    :raises StreamFeatureNotSupported: if the stream has a feature that is not
+                                       supported on the receiving side.
+    :raises PropertyInvalid: if one or more of the specified properties is invalid
+                             or has an invalid type or value.
+    :raises NameInvalid: if the name of either snapshot is invalid.
+    :raises NameTooLong: if the name of either snapshot is too long.
+
+    .. note::
+        The ``origin`` is ignored if the actual stream is an incremental stream
+        that is not a clone stream and the destination filesystem exists.
+        If the stream is a full stream and the destination filesystem does not
+        exist then the ``origin`` is checked for existence: if it does not exist
+        :exc:`.DatasetNotFound` is raised, otherwise :exc:`.StreamMismatch` is
+        raised, because that snapshot can not have any relation to the stream.
+
+    .. note::
+        If ``force`` is `True` and the stream is incremental then the destination
+        filesystem is rolled back to a matching source snapshot if necessary.
+        Intermediate snapshots are destroyed in that case.
+
+        However, none of the existing snapshots must have the same name as
+        ``snapname`` even if such a snapshot were to be destroyed.
+        The existing ``snapname`` snapshot always causes :exc:`.SnapshotExists`
+        to be raised.
+
+        If ``force`` is `True` and the stream is a full stream then the destination
+        filesystem is replaced with the received filesystem unless the former
+        has any snapshots.
+        Those prevent the destination filesystem from being rolled back / replaced.
```

Comment:
I thought that it would be clearer if I used "those" in reference to "those snapshots", but probably that was not clear.
So, would it better to explicitly say "Those snapshots prevent" or "This prevents"?